canonical/mir
# Atomic kms hardware cursor support

Now with 100% more atomic API!

Still a bit broken:
1. If we don't commit on cursor move, the cursor becomes very choppy
2. If we attempt a blocking commit on cursor move, but use , cursor and application window movement become sluggish when dragging a window around.
3. If we attempt a nonblocking commit on cursor move, the kernel/hardware/?? can't keep up and complains a bunch (works best though!).
4. Crashes sometimes (might be related to the input thread calling `set_cursor`, attempting a commit, and the main thread attempting a commit at the same time?)
5. Cursor sometimes disappears when entering or leaving application windows (actual content, drop shadow doesn't affect it)

How to test:
1. Log out of your graphical session
2. Switch to a VT
3. Make sure hardware cursors initialize correctly, this can be done by  running any Mir example with the argument `--platform-display-libs=mir:atomic-kms`  and piping the output to `grep -i 'hardware cursor'`
4. If the hardware cursor initializes correctly, you can go on using your compositor as normal.
